numpy/core/arrayprint.py

```diff
@@ -349,66 +341,38 @@ def _get_format_function(data, precision, suppress_small, formatter):
     else:
         return formatdict['numpystr']()
 
-def _array2string(a, max_line_width, precision, suppress_small, separator=' ',
-                  prefix="", formatter=None):
+def _array2string(a, options, separator=' ', prefix=""):
 
-    if a.size > _summaryThreshold:
+    if a.size > options['threshold']:
         summary_insert = "..., "
         data = _leading_trailing(a)
     else:
         summary_insert = ""
-        data = ravel(asarray(a))
+        if a.shape == () and options.get('sign', '') == 'legacy':
+            data = asarray(a)
+        else:
+            data = ravel(asarray(a))
```

Comment:
Can we just ditch the `ravel` here altogether? I don't think any of the builtin formatters actually care.

Reply:
At least `IntegerFormat` does, since it used `np.maximum.reduce` to find the largest int:
```python
>>> x = np.array([[1, 2], [3, 4]])
>>> np.maximum.reduce(x)
array([3, 4])
>>> np.maximum.reduce(np.ravel(x))
4
```

Reply:
Changing that to `np.max` would fix that